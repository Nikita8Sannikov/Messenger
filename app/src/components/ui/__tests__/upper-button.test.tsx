import { render, screen } from "@testing-library/react";

import UpperButton from "@components/ui/upper-button";

describe("UpperButton", () => {
	test("renders with KeyboardArrowUpIcon", () => {
		render(<UpperButton />);
		const button = screen.getByRole("button");
		expect(button).toBeInTheDocument();
		expect(button.querySelector("svg")).toBeInTheDocument();
	});

	test("applies upper-button class", () => {
		render(<UpperButton />);
		const button = screen.getByRole("button");
		expect(button).toHaveClass("upper-button");
	});

	test("passes through other props correctly", () => {
		render(
			<UpperButton 
				variant="contained" 
				color="primary"
				disabled
			/>,
		);
		const button = screen.getByRole("button");
		expect(button).toHaveClass("MuiButton-contained");
		expect(button).toHaveClass("MuiButton-colorPrimary");
		expect(button).toBeDisabled();
	});

	test("handles click events", () => {
		const handleClick = jest.fn();
		render(<UpperButton onClick={handleClick} />);
		const button = screen.getByRole("button");
		button.click();
		expect(handleClick).toHaveBeenCalledTimes(1);
	});

	test("matches snapshot", () => {
		const { asFragment } = render(
			<UpperButton variant="contained" color="primary" />,
		);
		expect(asFragment()).toMatchSnapshot();
	});
}); 