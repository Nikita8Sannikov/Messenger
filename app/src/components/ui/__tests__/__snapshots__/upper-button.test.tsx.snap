// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpperButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary upper-button css-74d805-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
      data-testid="KeyboardArrowUpIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
